generate_markdown_from_doxygen_xml.py

```diff
@@ -0,0 +1,1031 @@
+        ## Member Data Documentation 
+        if len(self.protected_attribs)>0:
+            outputstring+='\n\n## Member Data Documentation\n\n'
+            for the_item in self.protected_attribs:
+                outputstring+=the_item.markdown()
+                
+        ## Field Documentation
+        if len(self.public_static_attributes)>0 or len(self.public_attributes)>0:
+            outputstring+='\n\n## Field Documentation\n\n'
+            if len(self.public_static_attributes)>0:
+                for the_item in self.public_static_attributes:
+                    outputstring+=the_item.markdown()
+                    
+            if len(self.public_attributes)>0:
+                for the_item in self.public_attributes:
+                    outputstring+=the_item.markdown()
+                
+            
+            
+        outputstring=cleanup_markdown_string(outputstring)
+            
+        return outputstring
+
+
+# Start actual execution
+parser = argparse.ArgumentParser(description='Generate markdown from doxygen XML output.')
+parser.add_argument("-d", "--debug", default='',help="Any argument used here will include debug info in markdown output - ie some strings for various objects.")
+args = parser.parse_args()
+
+
+
+# Check if XML file path exists
+if not os.path.exists(DOXYGEN_ROOT_DIR):
+    print('Docs directory does not exist. Run: `make docs`')
+    exit()
+if not os.path.exists(DOXYGEN_XML_DIR):
+    print('XML directory does not exist (created by doxygen). Run: `make docs`')
+    exit()
+if not os.path.exists(DOXYGEN_OUTPUT_DIR):
+    #make output dir
+    os.makedirs(DOXYGEN_OUTPUT_DIR)
+
+
+
+skipped_files=[] # Files we have CHOSEN not to parse
+not_handled_files=[] # Files we don't handle, but maybe we should (catches new file types)
+for root, dirs, files in os.walk(DOXYGEN_XML_DIR, topdown=False):
+    for name in files:
+        current_filename = os.path.join(root, name)
+
+        if name[-4:]!='.xml':
+            skip_string=" %s - (not XML)" % current_filename
+            skipped_files.append(skip_string)
+            continue 
+        if name[-7:]=='_8h.xml':
+            skip_string=" %s - (header file)" % current_filename
+            skipped_files.append(skip_string)
+            continue 
+        if name[-9:]=='_8cpp.xml':
+            skip_string=" %s - (cpp file)" % current_filename
+            skipped_files.append(skip_string)
+            continue
+        if name[:4]=='dir_':
+            skip_string=" %s - (directory listing)" % current_filename
+            skipped_files.append(skip_string)
+            continue
+        if name=='index.xml':
+            skip_string=" %s - (index page)" % current_filename
+            skipped_files.append(skip_string)
+            continue
+        if name=='namespacedronecore.xml':
+            skip_string=" %s - (index page)" % current_filename
+            skipped_files.append(skip_string)
+            continue
+        if name[:5]=='class':
```

Comment:
You could probably do: `if 'class' in name:`.

Reply:
Maybe. Not changing this for now.